benchexec/tools/vampire.py

```diff
@@ -32,15 +32,15 @@ def environment(self, executable):
 
     def cmdline(self, executable, options, task, rlimits):
         if "-t" not in options and "--time_limit" not in options:
-            if rlimits.walltime:
+            if rlimits.walltime is None:
                 # Default timeout of Vampire is 60s,
                 # so we set value of 0 explicitly (means unlimited)
                 options += ["-t", "0"]
             else:
                 options += ["-t", f"{rlimits.walltime}s"]
 
         if "-m" not in options and "--memory_limit" not in options:
-            if rlimits.memory:
```

Comment:
Here I would suggest to just keep the `if rlimits.memory:` branch and omit the empty branch without limit (but keep the comment).